dvc/fs/http.py

```diff
@@ -190,6 +190,15 @@ def _upload(
                 desc=name or to_info.url,
             )
 
+    def open(self, path_info, mode: str = "r", encoding: str = None, **kwargs):
+        return super().open(
+            path_info,
+            mode,
+            encoding,
+            auth=self._auth_method(path_info),
+            **kwargs,
+        )
```

Comment:
```suggestion
            return open_url(
                path_info.url,
                mode=mode,
                encoding=encoding,
                auth=self._auth_method(path_info),
                **kwargs,
            )
```

Reply:
And we can remove `if hasattr(self, "_generate_download_url"):` block from `BaseFileSystem.open` and we can also remove `_generate_download_url` everywhere (http was the only one still using it)